client/src/main/java/org/asynchttpclient/netty/channel/ChannelManager.java

```diff
@@ -391,8 +387,49 @@ public SslHandler addSslHandler(ChannelPipeline pipeline, Uri uri, String virtua
         return sslHandler;
     }
 
-    public Bootstrap getBootstrap(Uri uri, ProxyServer proxy) {
-        return uri.isWebSocket() && proxy == null ? wsBootstrap : httpBootstrap;
+    public Promise<Bootstrap> getBootstrap(Request request, ProxyServer proxy) {
+        final Promise<Bootstrap> promise = ImmediateEventExecutor.INSTANCE.newPromise();
+
+        if (request.getUri().isWebSocket() && proxy == null) {
+            promise.setSuccess(wsBootstrap);
+        } else {
+            if (proxy != null && proxy.getProxyType().isSocks()) {
+                request.getNameResolver().resolve(proxy.getHost()).addListener((Future<InetAddress> proxyDnsFuture) -> {
+                    if (proxyDnsFuture.isSuccess()) {
+                        Bootstrap bootstrap = httpBootstrap.clone();
+                        ChannelHandler handler = bootstrap.config().handler();
```

Comment:
Correct me if I'm wrong, but it looks to me that you're editing the shared `httpBootstrap`, so it would stop wrong for other requests that wouldn't have this proxy configured.
Shouldn't installing the socks proxy handler be done in `NettyConnectListener`?